src/blockchain/log-processors/profit-loss/frozen-funds.ts

```diff
@@ -0,0 +1,99 @@
+// for every outcome on a market when a user claims their winnings for
+// that market. In practical terms, the user claims their winnings which
+// makes them have zero frozen funds for all outcomes in this market.
+type ClaimProceeds = "ClaimProceeds";
+
+// Trade is a type of FrozenFundsEvent corresponding to the user
+// executing a trade on a specific market outcome. An outcome's frozen
+// funds may increase or decrease depending on the details of the trade.
+interface Trade {
+  creatorOrFiller: "creator" | "filler"; // "creator" if the user was the creator of the Order to which this Trade belongs. "filler" if the user filled another creator's Order
+  realizedProfit: BigNumber; // denominated in tokens (eg. ETH). Profit which the user realized by executing this trade
+  tradeData: Pick<TradesRow<BigNumber>, "numCreatorTokens" | "numCreatorShares" | "numFillerTokens" | "numFillerShares">; // data associated with this Trade which is required to compute the next FrozenFunds
+}
+
+interface FrozenFundsParams {
+  frozenFundsBeforeEvent: FrozenFunds; // FrozenFunds prior to processing this FrozenFundsEvent
+  event: FrozenFundsEvent; // FrozenFundsEvent to process and return updated FrozenFunds as impacted by this event
+}
+
+// TODO doc - frozen funds in context of one (market, outcome)
+// TODO unit test
+export function getFrozenFundsAfterEventForOneOutcome(params: FrozenFundsParams): FrozenFunds {
+  if (params.event === "ClaimProceeds") {
+    return {
+      // TODO doc
+      frozenFunds: ZERO,
+      frozenProfit: ZERO,
+    };
+  }
+  // params.event is a trade executed; TODO break each event into a separate function
+  const trade = params.event.tradeData;
+  const frozenFundsAfterEvent = Object.assign({}, params.frozenFundsBeforeEvent);
+
+  // TODO make this into a helper
+  /*
+  function getTokensForDestroyedSharesForTrade(): {
+    creatorTokens: BigNumber,
+    fillerTokens: BigNumber,
+  } {
+    return {
+      creatorTokens: ZERO,
+      fillerTokens: ZERO,
+    };
+  }
+  */
+  const tokensFreedFromEscrowByThisTrade = BigNumber.min(trade.numCreatorShares, trade.numFillerShares).multipliedBy(ZERO /* TODO displayRange */);
+
+  // TODO Alex - can you confirm formula for this:
+  const creatorPortionOfTokensFreedFromEscrow = tokensFreedFromEscrowByThisTrade; // TODO actual value I think is f(minPrice, maxPrice, price)?
+  const fillerPortionOfTokensFreedFromEscrow = tokensFreedFromEscrowByThisTrade; // TODO actual value I think is f(minPrice, maxPrice, price)?
+
+  if (params.event.creatorOrFiller === "creator") {
+    // I created this trade
+    // TODO doc formulas
+    frozenFundsAfterEvent.frozenFunds = frozenFundsAfterEvent.frozenFunds.minus(creatorPortionOfTokensFreedFromEscrow);
+    frozenFundsAfterEvent.frozenFunds = frozenFundsAfterEvent.frozenFunds.plus(trade.numCreatorTokens);
+    frozenFundsAfterEvent.frozenFunds = frozenFundsAfterEvent.frozenFunds.minus(trade.numFillerTokens);
```

Comment:
Realizing now this won't quite work since the goal is to figure out how many tokens the traders actually received and this won't always coincide with that.

I think the best way may be to actually look at shares provided and the price.

So For tokens spent you can still use `frozenFunds = .frozenFunds.plus(trade.numCreatorTokens);`

but then we can avoid the complete set calc entirely and the above line and instead just do something like:

`tokensReceived = trade.numCreatorShares.multipliedBy(price);
frozenFunds = frozenFunds.minus(tokensReceived);`

where price will be `maxPrice - price` for the party shorting (whichever that is)